src/core/Clipboard.cpp

```diff
@@ -22,21 +22,83 @@
  *
  */
 
+#include "Clipboard.h"
+
+#include <set>
+
 #include <QApplication>
 #include <QClipboard>
-#include <QMimeData>
-
-#include "Clipboard.h"
 
+#include "FileBrowser.h"
+#include "PluginFactory.h"
+#include "SampleDecoder.h"
+#include "StringPairDrag.h"
 
 namespace lmms::Clipboard
 {
 
-	const QMimeData * getMimeData()
+	// there are other mimetypes, such as "samplefile", "patchfile" and "vstplugin" but they are generated dynamically.
+	static std::map<std::string, std::set<std::string>> mimetypes =
+	{
+		{"trackpresetfile", {"xpf", "xml"}},
+		{"midifile", {"mid", "midi", "rmi"}},
+		{"projectfile", {"mmp", "mpt", "mmpz"}},
+	};
+
+	//! gets the extension of a file, or returns the string back if no extension is found
+	static QString getExtension(const QString& file)
+	{
+		QFileInfo fi(file);
+		const QString ext = fi.suffix().toLower();
+		return ext.isEmpty() ? file.toLower() : ext;
+	}
+
+	/* @brief updates the extension map.
+	 */
+	void updateExtensionMap()
+	{
+		for (auto& pluginInfo : PluginFactory::instance()->pluginInfos())
+		{
+			const char* mimetype = pluginInfo.descriptor->supportedMimetype;
+
+			if (mimetype == nullptr || mimetype[0] == '\0') { continue; }
+
+			auto& existingTypes = mimetypes[mimetype]; // creates key if not present
+
+			const auto fileTypes = QString(pluginInfo.descriptor->supportedFileTypes).split(",");
+			for (auto& fileType : fileTypes)
+			{
+				existingTypes.insert(fileType.toStdString());
+			}
+		}
+	}
+
+
+	bool isType(const QString& ext, const QString& mimetype)
 	{
-		return QApplication::clipboard()->mimeData( QClipboard::Clipboard );
+		auto it = mimetypes.find(mimetype.toStdString());
+		if (it == mimetypes.end()) { return false; }
+
+		const auto& fileTypes = it->second;
+		const auto extStr = getExtension(ext).toStdString();
+
+		return fileTypes.find(extStr) != fileTypes.end();
 	}
 
+	bool isAudioFile(const QString& ext)		{ return isType(ext, "samplefile"); }
+	bool isProjectFile(const QString& ext)		{ return isType(ext, "projectfile"); }
+	bool isPluginPresetFile(const QString& ext)	{ return isType(ext, "pluginpresetfile"); }
+	bool isTrackPresetFile(const QString& ext)		{ return isType(ext, "trackpresetfile"); }
+	bool isSoundFontFile(const QString& ext)	{ return isType(ext, "soundfontfile"); }
+	bool isPatchFile(const QString& ext)		{ return isType(ext, "patchfile"); }
+	bool isMidiFile(const QString& ext)			{ return isType(ext, "midifile"); }
+	bool isVstPluginFile(const QString& ext)	{ return isType(ext, "vstpluginfile"); }
```

Comment:
```suggestion
	bool isAudioFile(const QString& ext)        { return isType(ext, "samplefile"); }
	bool isProjectFile(const QString& ext)      { return isType(ext, "projectfile"); }
	bool isPluginPresetFile(const QString& ext) { return isType(ext, "pluginpresetfile"); }
	bool isTrackPresetFile(const QString& ext)  { return isType(ext, "trackpresetfile"); }
	bool isSoundFontFile(const QString& ext)    { return isType(ext, "soundfontfile"); }
	bool isPatchFile(const QString& ext)        { return isType(ext, "patchfile"); }
	bool isMidiFile(const QString& ext)         { return isType(ext, "midifile"); }
	bool isVstPluginFile(const QString& ext)    { return isType(ext, "vstpluginfile"); }
```
Spaces should be used for alignment between the `)` and the `{` because tabs won't always display correctly.